src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs

```diff
@@ -129,7 +136,40 @@ public virtual string ToArtifact(object value, PropertyType propertyType, IColle
 
             value = JsonConvert.SerializeObject(blockEditorValue);
             _logger.Debug<BlockEditorValueConnector>("Finished converting {PropertyType} to artifact.", propertyType.Alias);
-            return (string) value;
+            return (string)value;
+        }
+
+        private Dictionary<string, IContentType> GetAllContentTypesUsedByBlocks(List<Block> allBlocks)
+        {
+            var allBlockContentTypeKeys = allBlocks
+                .Select(x => x.ContentTypeKey)
+                .Distinct()
+                .Select(ParseAsGuid)
+                .OrderBy(x => x)
+                .ToList();
+
+            // If we've requested this set of content types within the request, get from cache rather than going to the database again.
+            // This is useful as this method can be called multiple times in a save/publish operation.
+            var cacheKey = string.Format(DeployContribConstants.CacheKeys.DeployConnectorCacheContentTypesFormat, string.Join(",", allBlockContentTypeKeys));
+
+            // If we're on a background thread (e.g. for a content transfer), the request cache won't exist and we would fall through to the database retrieval.
+            // So instead we'll use the runtime cache (which will be cleared by Deploy between each background operation).
+            Func<Dictionary<string, IContentType>> getContentTypes = () =>
+                allBlockContentTypeKeys
+                    .ToDictionary(x => x.ToString(), x => _contentTypeService.Get(x));
```

Comment:
This still fetches the content types one by one from the database, which can be avoided by using `GetAll(allBlockContentTypeKeys)` instead.